src/java.desktop/windows/native/libawt/windows/awt_Robot.cpp

```diff
@@ -189,9 +191,21 @@ static void GetRGBPixels(jint x, jint y, jint width, jint height, jintArray pixe
     static const int BITS_PER_PIXEL = 32;
     static const int BYTES_PER_PIXEL = BITS_PER_PIXEL/8;
 
-    if (!IS_SAFE_SIZE_MUL(width, height)) throw std::bad_alloc();
+    if (!IS_SAFE_SIZE_MUL(width, height)) {
+        ::DeleteObject(hbitmap);
+        ::DeleteDC(hdcMem);
+        ::DeleteDC(hdcScreen);
+        throw std::bad_alloc();
+    }
+
     int numPixels = width*height;
-    if (!IS_SAFE_SIZE_MUL(BYTES_PER_PIXEL, numPixels)) throw std::bad_alloc();
+    if (!IS_SAFE_SIZE_MUL(BYTES_PER_PIXEL, numPixels)) {
+        ::DeleteObject(hbitmap);
+        ::DeleteDC(hdcMem);
+        ::DeleteDC(hdcScreen);
+        throw std::bad_alloc();
+    }
```

Comment:
I wonder if we can catch std::bad_alloc releases the resources and re-throw the exception.

Since a C++ exception is thrown, using try-catch for clean-up seems reasonable, this would avoid duplicating the clean-up code in three places or so.

Reply:
Hi Alexey,  probably we could do this. Do you think it is worth the effort (we would have a big try catch block instead, but would centralize the  ::Delete* calls) ?

Reply:
Since you've already integrated, it's not worth redoing in my opinion.

Error handling in C and C++ code is always prone to repetitive sequence of calls; in this particular case exceptions are already used, and it looks that handling that exception could have had a benefit of reducing code duplication.